modules/jta/src/main/java/org/jboss/weld/jta/TransactionalObserverNotifier.java

```diff
@@ -78,7 +82,19 @@ public ObserverNotifier create(String contextId, TypeSafeObserverResolver resolv
             for (ObserverMethod<? super T> observer : observers) {
                 deferNotification(event, metadata, observer, notifications);
             }
-            transactionServices.registerSynchronization(new TransactionNotificationSynchronization(notifications));
+            try {
+                transactionServices.registerSynchronization(new TransactionNotificationSynchronization(notifications));
+            } catch (Exception e) {
+                if (e.getCause() instanceof RollbackException || e.getCause() instanceof IllegalStateException) {
+                    List<ObserverMethod<? super T>> filteredObservers = observers.stream()
```

Comment:
We should sort the observers as desribed in https://github.com/cdi-spec/cdi/pull/272#issuecomment-205210486